### DISCUSSION TITLE
About FIle Upload component, how to display uploaded data

### DISCUSSION THREAD

**Discussion by tangbohao37** [chakra-ui/ark#3039] [Q&A] [answered]
After form submission, how can we display the data returned from the backend when revisiting the current page? For example, if the backend returns data such as an image URL, 'name', and 'size'. Is there a way to pass custom data into useFileUpload?

**Comment by cschroeter** [comment, marked as answer]
@tangbohao37  

Hey mate, thanks for reaching out! The `FileUpload` component uses the [[File API](https://developer.mozilla.org/en-US/docs/Web/API/File_API)](https://developer.mozilla.org/en-US/docs/Web/API/File_API), so the data is read-only and can't be manipulated directly. Instead, you should render a different output based on the data you receive from your server.
